### PR TITLE
GH-3087 default SHACL validation report size limits

### PR DESCRIPTION
**Pull request by hmottestad**
GitHub issue resolved: GH-3087 

Briefly describe the changes proposed in this PR:

 - set reasonable default limits for the SHACL validation report

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md) for more details):

 - [ ] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [ ] I've added tests for the changes I made
 - [ ] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [ ] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [ ] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

File: core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/config/ShaclSailConfig.java [hunk 1]
```diff
@@ -51,8 +51,8 @@
 	public static final boolean SERIALIZABLE_VALIDATION_DEFAULT = true;
 	public static final boolean ECLIPSE_RDF4J_SHACL_EXTENSIONS_DEFAULT = false;
 	public static final boolean DASH_DATA_SHAPES_DEFAULT = false;
-	public final static long VALIDATION_RESULTS_LIMIT_TOTAL_DEFAULT = -1;
-	public final static long VALIDATION_RESULTS_LIMIT_PER_CONSTRAINT_DEFAULT = -1;
+	public final static long VALIDATION_RESULTS_LIMIT_TOTAL_DEFAULT = 1_000_000;
+	public final static long VALIDATION_RESULTS_LIMIT_PER_CONSTRAINT_DEFAULT = 1_000;
```
Comment (hmottestad) [inline core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/config/ShaclSailConfig.java:55]: You guys think that these are reasonable limits?

Comment (jeenbroekstra) [reply core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/config/ShaclSailConfig.java:55]: I'm not sure - if you had to do a "back of the envelope" calculation of how much memory such a truncated report would consume at most - any clue? Put another way: how large is a single validation result?

Comment (hmottestad) [reply core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/config/ShaclSailConfig.java:55]: I can give it a shot.

Comment (jeenbroekstra) [reply core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/config/ShaclSailConfig.java:55]: Fwiw I'm fine with putting these limits in if you're confident these are reasonable - I just have no data on which to decide for myself how big 1 million validation results can get :)

Comment (hmottestad) [reply core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/config/ShaclSailConfig.java:55]: 1 000 000 violations needs around 700 MB of RAM (eg. setValidationResultsLimitTotal(0) saves 700 MB of RAM).

**Review by jeenbroekstra — APPROVED** [review]

**Pull request merged**